tests/queries/0_stateless/01526_client_start_and_exit.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "$CURDIR"/../shell_config.sh
+
+# Create a huge amount of tables, so Suggest will take a time to load
+seq 1 1000 | sed -r -e 's/(.+)/CREATE TABLE IF NOT EXISTS test\1 (x UInt8) ENGINE = Memory;/' | ${CLICKHOUSE_CLIENT} -n
```

Comment:
maybe try longer / random-like table/column names?

Reply:
Checked, it didn't help.

It seems it start reporducing on really big number of tables (maybe it makes SELECT which feed the suggestion list slow?)

Check this out:
```
root@27e49329c2e8:/# clickhouse-client -q "SELECT 'CREATE TABLE test_' || hex(randomPrintableASCII(40)) || '(x UInt8) Engine=Memory;' from numbers(200000)" --format=TSVRaw | clickhouse-benchmark -c32 -i 200000 -d 0
Loaded 200000 queries.

Queries executed: 200000 (100.000%).

localhost:9000, queries 200000, QPS: 3528.722, RPS: 0.000, MiB/s: 0.000, result RPS: 0.000, result MiB/s: 0.000.

0.000%		0.001 sec.	
10.000%		0.002 sec.	
20.000%		0.002 sec.	
30.000%		0.002 sec.	
40.000%		0.003 sec.	
50.000%		0.003 sec.	
60.000%		0.003 sec.	
70.000%		0.004 sec.	
80.000%		0.004 sec.	
90.000%		0.026 sec.	
95.000%		0.053 sec.	
99.000%		0.089 sec.	
99.900%		0.205 sec.	
99.990%		0.346 sec.	


root@27e49329c2e8:/# clickhouse-client 
ClickHouse client version 20.9.3.45 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.9.3 revision 54439.

27e49329c2e8 :) Bye.
Aborted (core dumped)
root@27e49329c2e8:/# clickhouse-client 
ClickHouse client version 20.9.3.45 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.9.3 revision 54439.

27e49329c2e8 :) Bye.
Aborted (core dumped)
root@27e49329c2e8:/# clickhouse-client 
ClickHouse client version 20.9.3.45 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.9.3 revision 54439.

27e49329c2e8 :) Bye.
Aborted (core dumped)
root@27e49329c2e8:/# clickhouse-client 
ClickHouse client version 20.9.3.45 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.9.3 revision 54439.

27e49329c2e8 :) Bye.
Aborted (core dumped)
```

Reply:
In my case it seems it starts reproducing when you have > 20K tables, and reproduces quite stable when I have about 50K of tables and hit Ctrl+D fast enough.

```
clickhouse-client -q "SELECT 'CREATE TABLE test_' || toString(number) || '(x UInt8) Engine=Memory;' from numbers(50000)" --format=TSVRaw | clickhouse-benchmark -c32 -i 50000 -d 0
```

Reply:
@filimonov You did not test carefully*,
this patch fixes the issue.

\* maybe forgot to replace `clickhouse-client` to the new build.

Reply:
Of course, i was checking a build w/o the fix applied - you just was saying you were not able to reproduce the problem, so i tried to explain a reproduce scenario better.

Fix looks good, and definitely should solve the issue.